docs/source/architecture/journal.rst

```diff
@@ -0,0 +1,379 @@
+BlockStore and need to be removed. After the BlockStore is updated, the Block
+Publisher is notified that there is a new chain head.
+
+Delayed Block Processing
+------------------------
+
+While the ChainController does Block validation in parallel, there are cases
+where the ChainController will serialize Block validation. These cases are when
+a Block is received and any of its predecessors are still being validated. In
+this case the validation of the predecessor is completed before the new block is
+scheduled. This is done to avoid redoing the validation work of the predecessor
+Block, since the predecessor must be validated prior to the new Block, the delay
+is inconsequential to the outcome.
+
+The BlockValidator
+------------------
+
+The BlockValidator is a subcomponent of the ChainController that is responsible
+for Block validation and fork resolution. When the BlockValidator is
+instantiated, it is given the candidate Block to validate and the current chain
+head.
+
+During processing, if a Block is marked as invalid it is discarded, never to be
+considered again. The only way to have the Block reconsidered is by flushing the
+BlockCache, which can be done by restarting the validator.
+
+The BlockValidator has three stages of evaluation. 
+
+1. Determine the common root of the fork(ForkRoot). This is done by walking the
+   chain back from the candidate and the chain head until a common block is
+   found. The Root can be the ChainHead in the case that the Candidate is
+   advancing the existing chain. The only case that the ForkRoot will not be
+   found is if the Candidate is from another Genesis. If this is the case, the
+   Candidate and all of its predecessors are marked as Invalid and discarded.
+   During this step, an ordered list of both chains is built back to the
+   ForkRoot.
+2. The Candidate chain is validated. This process walks forward from the
+   ForkRoot and applies block validation rules(described below) to each Block
+   successively. If any block fails validation, it and all of its successors
+   are marked as Invalid (Valid Blocks are defined as having Valid
+   predecessor(s)). Once the Candidate is successfully Validated and marked as
+   Valid, the Candidate is ready for Fork Resolution.
+3. Fork resolution requires a determination to be made if the Candidate should
+   replace the ChainHead. Fork resolution is a role of the Consensus algorithm
+   and is deferred entirely to the consensus implementation. Once the Consensus
+   determines if the block is the new ChainHead the answer is returned to the
+   ChainController which updates the BlockStore if the answer is in the
+   affirmative. Additionally, if the Candidate is to become the ChainHead, the
+   list of transactions committed in the new chain back to the common root is
+   computed and the same list is computed on the current chain. This
+   information is for to help the BlockPublisher update its pending batch list
+   when the chain is updated.
```

Comment:
These last sentence is very confusingly worded. What is computed, why?

Also:
"is ~for~ to help the BlockPublisher"